frontend/src/components/ViewActivities/activitylist.js

```diff
@@ -1,21 +1,73 @@
 import React from 'react';
-import app from '../Firebase';
 import * as activityFns from './activityfns.js';
 import ActivityDay from './activityday.js';
-import Accordion from 'react-bootstrap/Accordion';
 import '../../styles/activities.css';
+import * as DB from '../../constants/database.js';
+import app from '../Firebase';
+
+const db = app.firestore();
+
+
+/**
+ * Gets the list of activities from the server. 
+ * 
+ * @param {string} tripId The trip ID.
+ */
+export async function getActivityList(tripId) {
+  return new Promise(function(resolve, reject) {
+    let tripActivities = [];
+    
+    db.collection(DB.COLLECTION_TRIPS).doc(tripId)
+    .collection(DB.COLLECTION_ACTIVITIES).get()
+    .then(querySnapshot => {
+      querySnapshot.forEach(doc => {
+        let data = doc.data();
+        data['id'] = doc.id;
+        
+        // TODO: if start date != end date, split into 2 days. (#37)
 
+        // Eliminate nanoseconds, convert to milliseconds.
+        data[DB.ACTIVITIES_START_TIME] =
+          data[DB.ACTIVITIES_START_TIME]['seconds'] * 1000;         
+        data[DB.ACTIVITIES_END_TIME] = 
+          data[DB.ACTIVITIES_END_TIME]['seconds'] * 1000;
+
+        tripActivities.push(data);
+      })
+    }).catch(error => {
+      console.log("It seems that an error has occured.");
+      tripActivities = null;
+    }).then( () => resolve(tripActivities) );
+  })
+}
+
+/**
+ * The list of activities. 
+ * 
+ * @param {Object} props This component expects the following props:
+ * - `tripId` {string} The trip's ID.  
```

Comment:
Is this still true even though it doesn't appear used in this function anymore?

Reply:
This is a description for the class (that's how we decided we'd do these) and props.tripId is used in the class.